### PR TITLE
fix(samples): s2s java example and improve docs

### PR DESCRIPTION
**Pull request by efgpinto**
Small fix for the s2s java sample and some docs improvement.

### CODE REVIEW

**Review by octonato — APPROVED** [review]

File: samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java [hunk 1]
```diff
@@ -6,13 +6,13 @@
 
 public sealed interface CustomerEvent {
 
-  @TypeName("customer-created") // <1>
+  @TypeName("internal-customer-created") // <1>
```
Comment (aludwiko) [inline samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java:9]: Why the prefix `internal` ?

Comment (efgpinto) [reply samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java:9]: There is another sealed interface used for the public api.. and they had the same names. That causes an exception as these need to be unique. So I changed the internal ones to have that prefix.

Comment (octonato) [reply samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java:9]: I'm curious to understand why they need to be unique. 

Are they handle in different code paths? Why the name clash?

Comment (octonato) [reply samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java:9]: Oh yes, that's possible. Doesn't matter which components and how to get there. We always pass by the cache. 

Which makes me think if we should not have a namespace for the cache. A cache per component. For the user, there is not reason why they could not reuse a name. And can even be confusing when seeing such an error message. Worst, that's a runtime error.

Comment (efgpinto) [reply samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java:9]: here it is https://github.com/lightbend/kalix-jvm-sdk/issues/1556

**Pull request merged**